### DISCUSSION TITLE
Is there any apis like watch.immediate in vue or Model.setup in redux for subscribe?

### DISCUSSION THREAD

**Discussion by S-TE11A** [pmndrs/valtio#780] [Q&A]
# Summary
I want to trigger the callback of subscribeKey immediately as soon as entering the page (the proxy finished initialization.) so that I can value some other parameters with initial value. What I want is something like watch.immediate in vue or Model.setup in redux, but I cant find any way to do this in valtio. Whats the proper way to achieve this? 

sorry 4 my poor English and thank you for help!

**Comment by dai-shi** [comment]
Do you mean this? https://github.com/pmndrs/valtio/blob/4f4404d32907e0c62c23738e8fb21c86d3743f16/src/vanilla/utils/subscribeKey.ts#L18

Or, if it's about the initialization, you can do whatever you want after the initialization.
```js
const state = proxy({ foo: 'hey' })
callback(state.foo)
```

**Reply by S-TE11A** [reply]
I think you misunderstood my question. What I want to do is to find a proper way to trigger the callback immediately on subscribeKey creation.

**Reply by dai-shi** [reply]
Oops, my mistake on the second example. 

```ts
subscribeKey(state, "foo", callback)
callback(state.foo)
```
